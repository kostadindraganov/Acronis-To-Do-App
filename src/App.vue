<template>
  <el-container id="app">
    <el-header height="70px" class="custom-shadow">
      <HeaderMenu/>
    </el-header>
    <el-main>
      <router-view/>
    </el-main>
  </el-container>
</template>

<script>
  import HeaderMenu from "@/components/headerMenu";

  export default {
    name: "app",
    components: { HeaderMenu }
  };
</script>

