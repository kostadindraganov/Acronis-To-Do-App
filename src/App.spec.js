import { createLocalVue, mount, RouterLinkStub } from "@vue/test-utils";
import ElementUI from "element-ui";
import Vuex from "vuex";
import App from "@/App";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(ElementUI);
localVue.use(VueRouter);
const router = new VueRouter();

describe("App.vue", () => {
	let store;
	beforeEach(() => {
		store = new Vuex.Store({
			state: {
				route: {
					name: "home"
				}
			}
		});
	});

	it("Check header menu is link to about page", () => {
		const wrapper = mount(App, {
			localVue,
			router,
			store,
			stubs: {
				RouterLink: RouterLinkStub
			}
		});
		expect(wrapper.find(RouterLinkStub).props().to).toEqual({ name: "about" });
	});

	test("App.vue renders correctly", () => {
		const wrapper = mount(App, {
			localVue,
			store,
			router,
			stubs: {
				RouterLink: RouterLinkStub
			}
		});
		expect(wrapper.element).toMatchSnapshot();
	});
});
