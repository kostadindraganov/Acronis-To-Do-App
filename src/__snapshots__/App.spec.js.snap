// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue App.vue renders correctly 1`] = `
<section
  class="el-container is-vertical"
  id="app"
>
  <header
    class="el-header custom-shadow"
    style="height: 70px;"
  >
    <div
      class="header-menu el-row is-align-middle el-row--flex"
      height="70px"
    >
      <div
        class="el-col el-col-3"
      >
        <a>
          <button
            class="el-button el-button--text el-button--medium"
            type="button"
          >
            <!---->
            <!---->
            <span>
              About
            </span>
          </button>
        </a>
      </div>
    </div>
  </header>
   
  <main
    class="el-main"
  />
</section>
`;
